schema/bom-1.6.schema.json

```diff
@@ -3813,6 +3845,678 @@
+        "assetType": {
+          "type": "string",
+          "title": "assetType",
+          "description": "Cryptographic assets occur in several forms. Algorithms and protocols are most commonly implemented in specialized cryptographic libraries. They may however also be 'hardcoded' in software components. Certificates and related cryptographic material like keys, tokens, secrets or passwords are other cryptographic assets to be modelled.\n\n* __algorithm__ = Mathematical functions that encrypt text. They are used for data encryption, authentication, and digital signatures.\n* __certificate__ = An electronic document that is used to provide the identity or validate a public key.\n* __protocol__ = A set of rules and guidelines that govern the behavior and communication with each other.\n* __relatedCryptoMaterial__ = Other cryptographic assets that are related to algorithms, certificate, and protocols such as keys and tokens.",
+          "enum": [
+            "algorithm",
+            "certificate",
+            "protocol",
+            "relatedCryptoMaterial"
+          ]
+        },
+        "algorithmProperties": {
+          "type": "object",
+          "title": "Algorithm Properties",
+          "description": "Additional properties specific to a cryptographic algorithm.",
+          "additionalProperties": false,
+          "properties": {
+            "primitive": {
+              "type": "string",
+              "title": "primitive",
+              "description": "Low-level cryptographic building block used in higher-level cryptographic functions. Primitives can be used to construct algorithms used for encryption (e.g. AES), digital signatures (e.g. ECDSA), and pseudorandom number generators (e.g. PRNG).\n\n TODO Basil",
+              "enum": [
+                "drbg",
+                "mac",
+                "blockcipher",
+                "streamcipher",
+                "signature",
+                "hash",
+                "pke",
+                "xof",
+                "kdf",
+                "keyagree",
+                "kem",
+                "ae",
+                "other",
+                "unknown"
+              ]
+            },
+            "parameterSetIdentifier": {
+              "type": "string",
+              "title": "Parameter Set Identifier",
+              "description": "An identifier for the parameter set of the cryptographic algorithm. Examples: in AES128, '128' identifies the key length in bits, in SHA256, '256' identifies the digest length, '128' in SHAKE128 identifies its maximum security level in bits, and 'SHA2-128s' identifies a parameter set used in SLH-DSA (FIPS205)."
+            },
+            "curve": {
+              "type": "string",
+              "title": "Elliptic Curve",
+              "description": "The specific underlying Elliptic Curve (EC) definition employed which is an indicator of the level of security strength, performance and complexity.",
+              "enum": [
```

Comment:
better keep this an extra enum. exra as in enother file, like the SPDX ids are handled already.

this way we could update the list of enums, even no new release of CycloneDX happened.

Reply:
@stevespringett @bhess  toughts? 
Are there even other enum values that shoulb be managed extra?

Reply:
We are currently discussing the role of enums in the spec. We might externalize the curves (similar to SPDX license IDs), and we're working with a possible authoritative source for the curves.

Reply:
fixed via https://github.com/CycloneDX/specification/pull/347/commits/800ad2a7e4620ee5cbdab631c839219fc1c1cf8e